File: src/google/others/isomorphic_strings.cc
#include <bits/stdc++.h>
using namespace std;

class Solution {
  public:
    bool isIsomorphic(string s, string t) {
        unordered_map<char, char> converter;
        int n = s.size();
        for(int i = 0; i < n; i++) {
            if(converter.find(s[i]) == converter.end()) {
                converter[s[i]] = t[i];
            } else {
                if(converter[s[i]] != t[i])
                    return false;
            }
        }
        set<char> count;
        for(auto e : converter) {
			count.insert(e.second);
        }
		if(count.size() != converter.size())
			return false;
        return true;
    }
};

File: src/lc75/isomorphic_strings.py
class Solution:
    def isIsomorphic(self, s: str, t: str) -> bool:
        return self.is_surjective(s, t) and self.is_surjective(t, s)
    
    def is_surjective(self, s, t):
        mp = dict()
        for i, c in enumerate(s):
            if c in mp:
                if mp[c] != t[i]:
                    return False
            else:
                mp[c] = t[i]
        return True
